test/integration/canonicalize/canonical.expected

```diff
@@ -53,6 +53,39 @@ generated quantities {
   real y_hat[3, 2] = integrate_ode_rk45(sho, y0, 0.0, ts, theta, x_r, x_i);
 }
 
+
+Warning: deprecated language construct used in 'deprecated.stan', line 55, column 20 to column 69:
+integrate_ode is deprecated, use integrate_ode_rk45 instead
```

Comment:
Can we make these one line instead for ease of grep downstream? Maybe something like

```Warning: integrate_ode is deprecated and will be removed in the future. Use integrate_ode_rk45 instead``` 
which is close to stanc, or a one line version with line num if you prefer. Don't need to say "deprecated language construct" I think.

Reply:
Sure. Do you prefer the location information to be removed, as well? I thought it's quite helpful.

Reply:
I personally think the column information is superfluous, the line number seems plausibly helpful.

Reply:
I agree that the column information is not very important (in general imo); is there a way to print just line numbers? If not, and you think it's too cluttered as it is, I'll remove the Location_span part altogether.

Reply:
You should be able to access the line number as an integer from the start location inside the location span.

Reply:
I've implemented this, but it comes out (not surprisingly) too similar to [Location.to_string](https://github.com/stan-dev/stanc3/blob/30d1e55227fde791bd0177e997eb099de92aa675/src/middle/Location.ml#L44), which takes optional booleans for the file name and line number, but _not_ the column. Would it makes sense to add an optional `print_column` arg (defaults to true) to Location.to_string and call it, instead?

Reply:
> Would it makes sense to add an optional print_column arg (defaults to true) to Location.to_string and call it, instead?

To me that makes sense.

Reply:
A convention that I didn't know about until recently is that `to_blah` functions are generally supposed to be used for a total and reversible mapping between the type and a `blah`, e.g. if you will encode the entire `Location.t` in the string then `to_string` is appropriate, otherwise something like `line_string` might be better if you're just getting the line number formatted. Or just grabbing the line number out and doing the formatting where it's used, if you don't think that format will be shared. 

I know that's not how it currently works but maybe something to aim for going forward?

Reply:
I'd be happy to implement `line_string` if you think that's a good solution. Out of curiosity - Is there a place in which we call `Location.to_string` and we actually want the `column ... to column ... ` information? Is there a scenario in which the line number is not "good enough" for the user?

Reply:
I do think the column information is useful for reporting errors in long lines when we don't want to give the whole context window, e.g. in some of the pedantic mode warnings. That said:
- I'm in favor of adding that `print_column` option to `Location.to_string` (or something similarly named to follow Sean's convention)
- I'm also in favor of making the `column a to column b` much less verbose, I've been meaning to do that for a while. Alternatives might be `columns a:b` or `characters a:b`, some compilers even just do `a:b`

Reply:
So, do we want something like this:
```ocaml
let rec line_string ?(print_file = true) ?(print_line = true) ?(print_column = false) loc =
  let open Format in
  let file = if print_file then sprintf "'%s', " loc.filename else "" in
  let line = if print_line then sprintf "line %d, " loc.line_num else "" in
  let column = if print_column then sprintf "column %d, " loc.col_num else "" in
  let incl =
    match loc.included_from with
    | Some loc2 -> sprintf "included from\n%s" (line_string loc2)
    | None -> ""
  in
  sprintf "%s%s%s%s" file line column incl
```
If so, do we want to remove the optionals from `Location.to_string` (to enforce Sean's convention)? What about `Location_span.to_string`?

> I'm also in favor of making the column a to column b much less verbose, I've been meaning to do that for a while. Alternatives might be columns a:b or characters a:b, some compilers even just do a:b

I can do this, as well; I assume we need to somehow split to cases in which begin_loc and end_loc differ in their filename/included_from?

Reply:
Code looks good, but I think you'll need a little bit more case handling - for example, if `loc.included_from` is `None`, it'll have a trailing comma.

> If so, do we want to remove the optionals from Location.to_string (to enforce Sean's convention)? What about Location_span.to_string?

I'm happy either way w.r.t the naming. I do think whatever's decided for Location will probably also apply to Location_span.

> I can do this, as well; I assume we need to somehow split to cases in which begin_loc and end_loc differ in their filename/included_from?

That would be awesome. Yeah, I think that's right.

IMO this would be reasonable to do in a separate PR

Reply:
>Code looks good, but I think you'll need a little bit more case handling - for example, if loc.included_from is None, it'll have a trailing comma.

Is there a clean way to do this, and to avoid all the if-then-else scenarios? I've tried to pad the "%s%s%s%s" string with ", " according to the lengths of the different tags (hope that makes sense), but the type checker didn't like it...

Reply:
One way would be to build a list of strings and then use a library function like `String.join ~sep:", "`
One way to build that list would be to build a list of string options, then use List.filter_opt

Reply:
I see, so I can skip the whole sprintf part and build the string directly? That'd be easier if that's the case, I think.